### DISCUSSION TITLE
Is there a way to get Exiv2 running on c++17 and up?

### DISCUSSION THREAD

**Discussion by SteveXP** [Exiv2/exiv2#2599] [Q&A] [answered]
Exiv seams not to compile with c++17 because of autoPtr. This is probably relates to:
https://github.com/Exiv2/exiv2/issues/46

Is there any known workaround besides modifiing and recompiling exiv itselfe? 
Even the answer NO helps ;-)

**Comment by postscript-dev** [comment, marked as answer]
@SteveXP 
The current [maintenance branch](https://github.com/exiv2/exiv2/tree/0.27-maintenance) (for Exiv2 0.27.6) is fixed using C++98 and is not going to be updated.

However, the [main branch](https://github.com/exiv2/exiv2) supports C++20 and a new release is planned [any time now](https://github.com/Exiv2/exiv2/issues/2406#issuecomment-1529139799).

I hope this answers your question.
